app/src/test/java/com/codesoom/assignment/controllers/TaskControllerTest.java

```diff
@@ -0,0 +1,267 @@
+            void it_returns_task_list() {
+                List<Task> tasks = controller.list();
+                assertThat(tasks).isNotEmpty();
+                assertThat(tasks).hasSize((int) (createTaskSzie - deleteTaskSize));
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("Task 객체를 ID 로 조회하는 API는")
+    class Describe_detail_of_task {
+        static final long createTaskSize = 5L;
+
+        @BeforeEach
+        void setUp() {
+            createTask(createTaskSize);
+        }
+
+        @Nested
+        @DisplayName("ID 에 해당하는 Task가 존재하지 않을 경우")
+        class Context_with_invalid_id {
+            final long deleteTaskSize = 5L;
+            final long requestTaskId = 4L;
+
+            @BeforeEach
+            void setUp() {
+                for (long i = 1; i <= deleteTaskSize; i++) {
+                    controller.delete(i);
+                }
+            }
+
+            @Test
+            @DisplayName("TaskNotFoundException을 발생시킨다")
+            void it_throw_TaskNotFoundException() {
+                assertThatThrownBy(() -> controller.detail(requestTaskId))
+                        .isInstanceOf(TaskNotFoundException.class);
+            }
+        }
+
+        @Nested
+        @DisplayName("ID 에 해당하는 Task가 존재할 경우")
+        class Context_with_valid_id {
+            final long deleteTaskSize = 2L;
+            final long requestTaskId = 3L;
+
+            @BeforeEach
+            void setUp() {
+                for (long i = 1; i <= deleteTaskSize; i++) {
+                    controller.delete(i);
+                }
+            }
+            
+            @Test
+            @DisplayName("Task 객체를 리턴한다")
+            void it_return_task() {
+                Task foundTask = controller.detail(requestTaskId);
+
+                assertThat(foundTask).isNotNull();
+                assertThat(foundTask.getTitle()).isEqualTo(TASK_TITLE);
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("Task 객체를 ID 로 업데이트 하는 메소드는")
+    class Describe_update_of_task {
+        final long createTaskSize = 10L;
+
+        @BeforeEach
+        void setUp() {
+            createTask(createTaskSize);
+        }
+
+        @Nested
+        @DisplayName("ID 에 해당하는 Task 객체가 없을 경우")
+        class Context_with_invalid_id {
+            final long deleteTaskSize = 10L;
+            final long requestTaskId = 10L;
+
+            @BeforeEach
+            void setUp() {
+                for (long i = 1; i <= deleteTaskSize; i++) {
+                    controller.delete(i);
+                }
+            }
+
+            @Test
+            @DisplayName("TaskNotFoundException을 발생시킨다")
+            void it_throw_TaskNotFoundException() {
+                Task task = new Task();
+                task.setTitle(UPDATE_TASK_TITLE);
+
+                assertThatThrownBy(() -> controller.update(requestTaskId, task))
+                        .isInstanceOf(TaskNotFoundException.class);
+            }
```

Comment:
`예외 발생`은 적절한 표현이 아닐 수 있습니다. 애플리케이션을 사용하는 사람의 입장에서 예외는 발생하는 것이지만, 만드는 사람(우리)의 입장에서 예외는 "던지는" 것입니다. `throw` 키워드 단어의 뜻을 곰곰히 생각해 보세요. 그리고 아래쪽에 있는 `assertThatThrownBy`의 뜻도 열심히 생각해 보세요.

```suggestion
            @DisplayName("TaskNotFoundException을 던진다")
            void it_throw_TaskNotFoundException() {
                Task task = new Task();
                task.setTitle(UPDATE_TASK_TITLE);

                assertThatThrownBy(() -> controller.update(requestTaskId, task))
                        .isInstanceOf(TaskNotFoundException.class);
            }
```